tests/cpp/test_host_ir_evaluator.cpp

```diff
@@ -140,27 +145,93 @@ TEST_F(HostIrEvaluatorTest, AddInLoop) {
         IrBuilder::create<ShardByStream>(loop_out, out, stream_index);
     for_loop->body().push_back(shard_out);
 
-    // In reality, this should be a LaunchKernel and the two `ShardByStream`s
-    // should disappear. But currently we can't pass streamIdx to a kernel yet.
-    auto* add = IrBuilder::create<BinaryOp>(
-        BinaryOpType::Add, loop_out, loop_in, loop_in);
-    for_loop->body().push_back(add);
+    auto* mm = IrBuilder::create<MatmulOp>(loop_out, in, loop_w);
+    for_loop->body().push_back(mm);
```

Comment:
Can you add a comment here, that this `matmul` is not a launch kernel since we are using the ExprEval path making it a `top-level` expr?